File: SEP18/2.cs
using System;
using System.Linq;
					
public class ChefAdv {
	public static void Main() {
		// int n = Convert.ToInt32(Console.ReadLine());
		// int mid = Convert.ToInt32(n / 2);
		// for(int i = 2; i <= mid; i++) Console.Write(i + ' ');
		// Console.Write(1 + ' ');
		// for(int i = mid + 2; i < n + 1; i++) Console.Write(i + ' ');
		// Console.WriteLine(mid + 1);
		// Console.Write(n + ' ');
		// for(int i = 0; i < n + 1; i++) Console.Write(i + ' ');
		while(t-- > 0){
			int[] a = Console.ReadLine().Split(' ').Select(int.Parse).ToArray();
			int c = a[0] % a[2];
			int d = a[1] % a[3];
			c --;
			d --;
			if(c > 0 && d > 0){
				c--;
				d--;
			}
			if(c > 0 && d > 0){
				Console.WriteLine("Pofik");
			}else{
				Console.WriteLine("Chefirnemo");
			}
		}
		
	}
}

File: SEP18/2.py
for _ in range(int(input())):
	n,m,x,y = [int(x) for x in input().split()]
	f = 0
	a,b = (n - 1) % x, (m - 1) % y
	if (n == 1 and m == 1) or (n == 2 and m == 2) or (x == 1 and y == 1): f = 1
	elif (a == 0 and b == 0) or (a == 1 and b == 1): f = 1
	elif (a == 1 and m == 2) or (n == 2 and b == 1) or (a == 0 and m == 1) or (n == 1 and b == 0) or (x == 1 and (b == 0 or b == 1) and n > 1) or (y == 1 and (a == 0 or a == 1) and m > 1):f = 1
	print("Chefirnemo") if f == 1 else print("Pofik")